scipy/stats/_binned_statistic.py

```diff
@@ -645,7 +645,7 @@ def _calc_binned_statistic(Vdim, bin_numbers, result, values, stat_func,
             # if the stat_func is np.std, calc std only when binned data is 2
             # or more for speed up.
             if is_callable or not (stat_func is np.std and len(bin_map[i]) < 2):
-                result[vv, i] = stat_func(bin_map[i])
+                result[vv, i] = stat_func(np.array(bin_map[i]))
```

Comment:
If the `stats` folks do decide this is the right approach we could also add a short regression test based on the case reported in the original issue I suppose.

We probably wouldn't want to use 1 million elements though, but I imagine reducing the data size wouldn't be hard for the reproducer.

Reply:
Sure I can add a test. But is this actually needed. This is just converting into an array so that in the end we give an array to the user instead of a list.

I can add a small test for that and yes, no need at all to use 1M elements but a tiny number is enough to show this.

Reply:
Let's wait for a `stats` regular to comment before worrying about a test. My impression was that the user might face an error in some circumstances if they followed the docs and depended too strongly on the typing to an array.

Reply:
This change looks correct to me, we'd want to pass arrays rather than lists to the user-provided function.

This bit is fairly performance-critical, I'm not sure if this change will show up in a benchmark. However, I imagine that users would anyway convert to an array in most cases. So this looks fine to me.

@tupui maybe run the `BinnedStatisticDD` benchmark with this case? This case is already covered, by the `params = [..., np.std]` case.

Reply:
ok I will check, thanks.

Reply:
agreed, doesn't really need a test